cryptoki/src/types/session.rs

```diff
@@ -83,3 +105,105 @@ impl From<UserType> for CK_USER_TYPE {
         }
     }
 }
+
+/// Represents the state of a session
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct SessionState {
+    val: CK_STATE,
+}
+
+impl SessionState {
+    /// Read-only public session
+    pub const RO_PUBLIC_SESSION: SessionState = SessionState {
+        val: CKS_RO_PUBLIC_SESSION,
+    };
+
+    /// Read-only user access
+    pub const RO_USER_FUNCTIONS: SessionState = SessionState {
+        val: CKS_RO_USER_FUNCTIONS,
+    };
+
+    /// Read/write public session
+    pub const RW_PUBLIC_SESSION: SessionState = SessionState {
+        val: CKS_RW_PUBLIC_SESSION,
+    };
+
+    /// Read/write user access
+    pub const RW_USER_FUNCTIONS: SessionState = SessionState {
+        val: CKS_RW_USER_FUNCTIONS,
+    };
+
+    /// Read/write SO access
+    pub const RW_SO_FUNCTIONS: SessionState = SessionState {
+        val: CKS_RW_SO_FUNCTIONS,
+    };
+
+    fn state_to_str(&self) -> &'static str {
```

Comment:
That's a fair point.  Maybe it would make more sense to make this a static method that takes in a `CK_STATE` value and make it `pub` or `pub (struct)` so it's more generalized.